### ISSUE TITLE
GitHub and curl.se tarballs have different content

### ISSUE THREAD

**Issue by stevenpackardblp** [curl/curl#13184]
### I did this

I tried to apply https://github.com/curl/curl/commit/ed09a99af57200643d5ae001e815eeab9ffe3f84 as a patch to curl 8.6.0 to backport into our build and release system, which uses the tarball from curl.se published on https://curl.se/download.html. The commit failed to apply as a patch, because the test files in the commit do not exist in the curl-8.6.0.tar.gz from https://curl.se/download.html! They do exist in the release artifacts published to github.com: https://github.com/curl/curl/releases/tag/curl-8_6_0

### I expected the following

The release tarball on curl.se and GitHub would have the same code.

### curl/libcurl version

curl 8.6.0

### operating system

RedHat Enterprise 7.9

**Comment by dfandrich** [comment]
That sounds like #12914. Note also that the release tar balls contain some generated files, so the source at the git tag will not be exactly the same as the release version. See [GIT-INFO.md](/curl/curl/blob/master/GIT-INFO.md) and [maketgz](/curl/curl/blob/master/maketgz)

**Comment by bagder** [comment]
If we could, we would disable the GitHub generated ones.

See https://github.com/orgs/community/discussions/6003

**Comment by jay** [comment]
> That sounds like #12914

It is. I ran maketgz on master and test/http python files are included

**Comment by stevenpackardblp** [comment]
I feel this was closed pre-emptively. I don't believe this is a duplicate of #12914. This goes beyond generated files.

curl.se tarball has only these http test files:

```
$ tar -tzvf curl_8.6.0.orig.tar.gz |grep "tests/http"
drwxr-xr-x 1000/1000         0 2024-01-31 02:04 curl-8.6.0/tests/http/
drwxr-xr-x 1000/1000         0 2024-01-31 02:04 curl-8.6.0/tests/http/clients/
-rw-r--r-- 1000/126       7389 2023-12-08 08:25 curl-8.6.0/tests/http/clients/ws-data.c
-rw-r--r-- 1000/1000      9007 2024-01-23 10:21 curl-8.6.0/tests/http/clients/h2-download.c
-rw-r--r-- 1000/126       7385 2023-12-08 08:25 curl-8.6.0/tests/http/clients/h2-upgrade-extreme.c
-rw-r--r-- 1000/1000     30792 2024-01-29 10:53 curl-8.6.0/tests/http/clients/Makefile.in
-rw-r--r-- 1000/126       1222 2023-12-08 08:25 curl-8.6.0/tests/http/clients/Makefile.inc
-rw-r--r-- 1000/126       4195 2023-12-08 08:25 curl-8.6.0/tests/http/clients/ws-pingpong.c
-rw-r--r-- 1000/126       2282 2023-12-08 08:25 curl-8.6.0/tests/http/clients/Makefile.am
-rw-r--r-- 1000/1000      7283 2024-01-23 10:21 curl-8.6.0/tests/http/clients/h2-serverpush.c
-rw-r--r-- 1000/126       9654 2023-12-08 08:25 curl-8.6.0/tests/http/clients/h2-pausing.c
-rw-r--r-- 1000/126       8597 2023-12-08 08:25 curl-8.6.0/tests/http/clients/tls-session-reuse.c
-rw-r--r-- 1000/126       1181 2023-12-08 08:25 curl-8.6.0/tests/http/config.ini.in
-rw-r--r-- 1000/1000     24266 2024-01-29 10:53 curl-8.6.0/tests/http/Makefile.in
-rw-r--r-- 1000/126       7477 2023-12-08 08:25 curl-8.6.0/tests/http/README.md
-rw-r--r-- 1000/126       1191 2023-12-08 08:25 curl-8.6.0/tests/http/Makefile.am
-rwxr-xr-x 1000/126       5115 2023-12-08 08:25 curl-8.6.0/tests/http-server.pl
-rwxr-xr-x 1000/126       3394 2023-12-08 08:25 curl-8.6.0/tests/http2-server.pl
-rwxr-xr-x 1000/126       3374 2023-12-08 08:25 curl-8.6.0/tests/http3-server.pl
```

The GitHub tarball has these http test files:

```
$ tar -tzvf curl-8_6_0.tar.gz |grep "tests/http"
-rwxrwxr-x root/root      5115 2024-01-31 02:03 curl-curl-8_6_0/tests/http-server.pl
drwxrwxr-x root/root         0 2024-01-31 02:03 curl-curl-8_6_0/tests/http/
-rw-rw-r-- root/root       108 2024-01-31 02:03 curl-curl-8_6_0/tests/http/.gitignore
-rw-rw-r-- root/root      1191 2024-01-31 02:03 curl-curl-8_6_0/tests/http/Makefile.am
-rw-rw-r-- root/root      7477 2024-01-31 02:03 curl-curl-8_6_0/tests/http/README.md
drwxrwxr-x root/root         0 2024-01-31 02:03 curl-curl-8_6_0/tests/http/clients/
-rw-rw-r-- root/root       187 2024-01-31 02:03 curl-curl-8_6_0/tests/http/clients/.gitignore
-rw-rw-r-- root/root      2282 2024-01-31 02:03 curl-curl-8_6_0/tests/http/clients/Makefile.am
-rw-rw-r-- root/root      1222 2024-01-31 02:03 curl-curl-8_6_0/tests/http/clients/Makefile.inc
-rw-rw-r-- root/root      9007 2024-01-31 02:03 curl-curl-8_6_0/tests/http/clients/h2-download.c
-rw-rw-r-- root/root      9654 2024-01-31 02:03 curl-curl-8_6_0/tests/http/clients/h2-pausing.c
-rw-rw-r-- root/root      7283 2024-01-31 02:03 curl-curl-8_6_0/tests/http/clients/h2-serverpush.c
-rw-rw-r-- root/root      7385 2024-01-31 02:03 curl-curl-8_6_0/tests/http/clients/h2-upgrade-extreme.c
-rw-rw-r-- root/root      8597 2024-01-31 02:03 curl-curl-8_6_0/tests/http/clients/tls-session-reuse.c
-rw-rw-r-- root/root      7389 2024-01-31 02:03 curl-curl-8_6_0/tests/http/clients/ws-data.c
-rw-rw-r-- root/root      4195 2024-01-31 02:03 curl-curl-8_6_0/tests/http/clients/ws-pingpong.c
-rw-rw-r-- root/root      1181 2024-01-31 02:03 curl-curl-8_6_0/tests/http/config.ini.in
-rw-rw-r-- root/root      2749 2024-01-31 02:03 curl-curl-8_6_0/tests/http/conftest.py
-rw-rw-r-- root/root      1095 2024-01-31 02:03 curl-curl-8_6_0/tests/http/requirements.txt
-rw-rw-r-- root/root     25707 2024-01-31 02:03 curl-curl-8_6_0/tests/http/scorecard.py
-rw-rw-r-- root/root      4186 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_01_basic.py
-rw-rw-r-- root/root     19103 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_02_download.py
-rw-rw-r-- root/root      5949 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_03_goaway.py
-rw-rw-r-- root/root      6614 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_04_stuttered.py
-rw-rw-r-- root/root      4702 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_05_errors.py
-rw-rw-r-- root/root      5144 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_06_eyeballs.py
-rw-rw-r-- root/root     24405 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_07_upload.py
-rw-rw-r-- root/root      8359 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_08_caddy.py
-rw-rw-r-- root/root      3381 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_09_push.py
-rw-rw-r-- root/root     18187 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_10_proxy.py
-rw-rw-r-- root/root      4332 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_11_unix.py
-rw-rw-r-- root/root      2979 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_12_reuse.py
-rw-rw-r-- root/root      7368 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_13_proxy_auth.py
-rw-rw-r-- root/root      6150 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_14_auth.py
-rw-rw-r-- root/root      4139 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_15_tracing.py
-rw-rw-r-- root/root      5000 2024-01-31 02:03 curl-curl-8_6_0/tests/http/test_20_websockets.py
drwxrwxr-x root/root         0 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/
-rw-rw-r-- root/root      1509 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/__init__.py
-rw-rw-r-- root/root      5622 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/caddy.py
-rw-rw-r-- root/root     20747 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/certs.py
-rw-rw-r-- root/root      4236 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/client.py
-rw-rw-r-- root/root     26720 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/curl.py
-rw-rw-r-- root/root     15419 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/env.py
-rw-rw-r-- root/root     16556 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/httpd.py
drwxrwxr-x root/root         0 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/mod_curltest/
-rw-rw-r-- root/root        99 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/mod_curltest/.gitignore
-rw-rw-r-- root/root     17556 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/mod_curltest/mod_curltest.c
-rw-rw-r-- root/root      9702 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/nghttpx.py
-rw-rw-r-- root/root      1558 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/ports.py
-rwxrwxr-x root/root      1997 2024-01-31 02:03 curl-curl-8_6_0/tests/http/testenv/ws_echo_server.py
-rwxrwxr-x root/root      3394 2024-01-31 02:03 curl-curl-8_6_0/tests/http2-server.pl
-rwxrwxr-x root/root      3374 2024-01-31 02:03 curl-curl-8_6_0/tests/http3-server.pl
```

All the .py files and the entire testenv directory are notably missing from the curl.se tarball and are in source control.

**Comment by jay** [comment]
> All the .py files and the entire testenv directory are notably missing from the curl.se tarball and are in source control.

Sure, we understand what you are saying but the issue has been fixed since then. The next release will include the files.

**Comment by bagder** [comment]
It was fixed in #12917, you can double-check how the [daily snaphots](https://curl.se/snapshots/) look.

**Comment by stevenpackardblp** [comment]
Ah, sorry, I see #12917 did address this. Thanks!
